lib/components/pinnable/scroll_view.dart

```diff
@@ -70,8 +71,7 @@ class PinnableMessageScrollView extends ScrollView {
       }
       final pinned = isPinnedBuilder(nextPinnableIndex);
       final sliver = PinnableMessageSliver(
-        // key by index.
-        key: ValueKey(start),
+        key: key,
```

Comment:
Doesn't using the same key conflict when there is a pinned element?

Reply:
Oh good catch, I missed that the conditional didn't have a `continue`.